### PR TITLE
Merge fh/mymaster into master

### PR DESCRIPTION
**Pull request by flyinghead**

### CODE REVIEW

**Comment by skmp** [comment]
This pull request **introduces 184 alerts** and **fixes 25** when merging c809c6c56fab9558795d6bc0b00d19acd6013aff into 69ab16ad9c85a79f50df5974194b64e88e549e0c - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-39e3f34aa5efa298ef4cd97546c4bc4f3239be4d)

**new alerts:**

* 118 for FIXME comment
* 33 for Empty branch of conditional
* 10 for Multiplication result converted to larger type
* 9 for Comparison result is always the same
* 5 for Missing header guard
* 3 for Declaration hides parameter
* 1 for Unsigned comparison to zero
* 1 for Inconsistent definition of copy constructor and assignment \(&#39;Rule of Two&#39;\)
* 1 for Local variable hides global variable
* 1 for Use of potentially dangerous function
* 1 for Too few arguments to formatting function
* 1 for Short global name

**fixed alerts:**

* 19 for Wrong type of arguments to formatting function
* 3 for FIXME comment
* 1 for Missing header guard
* 1 for Function declared in block
* 1 for Too few arguments to formatting function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **introduces 184 alerts** and **fixes 25** when merging 22b08bea74fc1c4f96aec70ee31c443ed45a4ad0 into 69ab16ad9c85a79f50df5974194b64e88e549e0c - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-1d78754bd1bc887e921f586114476ddcb84dbb8d)

**new alerts:**

* 118 for FIXME comment
* 33 for Empty branch of conditional
* 10 for Multiplication result converted to larger type
* 9 for Comparison result is always the same
* 5 for Missing header guard
* 3 for Declaration hides parameter
* 1 for Unsigned comparison to zero
* 1 for Inconsistent definition of copy constructor and assignment \(&#39;Rule of Two&#39;\)
* 1 for Local variable hides global variable
* 1 for Use of potentially dangerous function
* 1 for Too few arguments to formatting function
* 1 for Short global name

**fixed alerts:**

* 19 for Wrong type of arguments to formatting function
* 3 for FIXME comment
* 1 for Missing header guard
* 1 for Function declared in block
* 1 for Too few arguments to formatting function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **introduces 184 alerts** and **fixes 25** when merging 815616369ef25b72983abce803b5820aabdb07de into 69ab16ad9c85a79f50df5974194b64e88e549e0c - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-c167b525eff6c06b63365bd05c3656233bdde3b3)

**new alerts:**

* 118 for FIXME comment
* 33 for Empty branch of conditional
* 10 for Multiplication result converted to larger type
* 9 for Comparison result is always the same
* 5 for Missing header guard
* 3 for Declaration hides parameter
* 1 for Unsigned comparison to zero
* 1 for Inconsistent definition of copy constructor and assignment \(&#39;Rule of Two&#39;\)
* 1 for Local variable hides global variable
* 1 for Use of potentially dangerous function
* 1 for Too few arguments to formatting function
* 1 for Short global name

**fixed alerts:**

* 19 for Wrong type of arguments to formatting function
* 3 for FIXME comment
* 1 for Missing header guard
* 1 for Function declared in block
* 1 for Too few arguments to formatting function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **introduces 184 alerts** and **fixes 25** when merging d6fd3e078100b933dd4ea895ed03ff1f37d9ad9d into 69ab16ad9c85a79f50df5974194b64e88e549e0c - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-1767199a5ec10c2675aec69726486394307d1de7)

**new alerts:**

* 118 for FIXME comment
* 33 for Empty branch of conditional
* 10 for Multiplication result converted to larger type
* 9 for Comparison result is always the same
* 5 for Missing header guard
* 3 for Declaration hides parameter
* 1 for Unsigned comparison to zero
* 1 for Inconsistent definition of copy constructor and assignment \(&#39;Rule of Two&#39;\)
* 1 for Local variable hides global variable
* 1 for Use of potentially dangerous function
* 1 for Too few arguments to formatting function
* 1 for Short global name

**fixed alerts:**

* 19 for Wrong type of arguments to formatting function
* 3 for FIXME comment
* 1 for Missing header guard
* 1 for Function declared in block
* 1 for Too few arguments to formatting function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **introduces 184 alerts** and **fixes 25** when merging cce5028ed3816faf1f25a4d1334028ef68e9e45f into 69ab16ad9c85a79f50df5974194b64e88e549e0c - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-c96f64953e2b305ca93834544e8f4dd4a8a6d9a7)

**new alerts:**

* 118 for FIXME comment
* 33 for Empty branch of conditional
* 10 for Multiplication result converted to larger type
* 9 for Comparison result is always the same
* 5 for Missing header guard
* 3 for Declaration hides parameter
* 1 for Unsigned comparison to zero
* 1 for Inconsistent definition of copy constructor and assignment \(&#39;Rule of Two&#39;\)
* 1 for Local variable hides global variable
* 1 for Use of potentially dangerous function
* 1 for Too few arguments to formatting function
* 1 for Short global name

**fixed alerts:**

* 19 for Wrong type of arguments to formatting function
* 3 for FIXME comment
* 1 for Missing header guard
* 1 for Function declared in block
* 1 for Too few arguments to formatting function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **introduces 184 alerts** and **fixes 25** when merging f5c896121e2ff72673698b56b0b398465c354f03 into 69ab16ad9c85a79f50df5974194b64e88e549e0c - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-9e7763686c829bfdecb5b3a79d734a2f2e08625d)

**new alerts:**

* 118 for FIXME comment
* 33 for Empty branch of conditional
* 10 for Multiplication result converted to larger type
* 9 for Comparison result is always the same
* 5 for Missing header guard
* 3 for Declaration hides parameter
* 1 for Unsigned comparison to zero
* 1 for Inconsistent definition of copy constructor and assignment \(&#39;Rule of Two&#39;\)
* 1 for Local variable hides global variable
* 1 for Use of potentially dangerous function
* 1 for Too few arguments to formatting function
* 1 for Short global name

**fixed alerts:**

* 19 for Wrong type of arguments to formatting function
* 3 for FIXME comment
* 1 for Missing header guard
* 1 for Function declared in block
* 1 for Too few arguments to formatting function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **introduces 185 alerts** and **fixes 25** when merging a3f97649360a52f347c03406fde3655decdd832e into 69ab16ad9c85a79f50df5974194b64e88e549e0c - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-722cbec5096454e05ccab75e873b92c76618cecf)

**new alerts:**

* 119 for FIXME comment
* 33 for Empty branch of conditional
* 10 for Multiplication result converted to larger type
* 9 for Comparison result is always the same
* 5 for Missing header guard
* 3 for Declaration hides parameter
* 1 for Unsigned comparison to zero
* 1 for Inconsistent definition of copy constructor and assignment \(&#39;Rule of Two&#39;\)
* 1 for Local variable hides global variable
* 1 for Use of potentially dangerous function
* 1 for Too few arguments to formatting function
* 1 for Short global name

**fixed alerts:**

* 19 for Wrong type of arguments to formatting function
* 3 for FIXME comment
* 1 for Missing header guard
* 1 for Function declared in block
* 1 for Too few arguments to formatting function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **introduces 185 alerts** and **fixes 26** when merging ca0816042e116c65a670853b056cd5e15ef14062 into 69ab16ad9c85a79f50df5974194b64e88e549e0c - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-67119d9f9f6216604b730dd1499444e539f240bb)

**new alerts:**

* 119 for FIXME comment
* 33 for Empty branch of conditional
* 10 for Multiplication result converted to larger type
* 9 for Comparison result is always the same
* 5 for Missing header guard
* 3 for Declaration hides parameter
* 1 for Unsigned comparison to zero
* 1 for Inconsistent definition of copy constructor and assignment \(&#39;Rule of Two&#39;\)
* 1 for Local variable hides global variable
* 1 for Use of potentially dangerous function
* 1 for Too few arguments to formatting function
* 1 for Short global name

**fixed alerts:**

* 20 for Wrong type of arguments to formatting function
* 3 for FIXME comment
* 1 for Missing header guard
* 1 for Function declared in block
* 1 for Too few arguments to formatting function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **introduces 185 alerts** and **fixes 26** when merging 1ef8caff882505d2fe6c88121353ba4b9a2e22b2 into 69ab16ad9c85a79f50df5974194b64e88e549e0c - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-8bdab0e27d2f3a6a4f419f3298274f0c03bb8287)

**new alerts:**

* 119 for FIXME comment
* 33 for Empty branch of conditional
* 10 for Multiplication result converted to larger type
* 9 for Comparison result is always the same
* 5 for Missing header guard
* 3 for Declaration hides parameter
* 1 for Unsigned comparison to zero
* 1 for Inconsistent definition of copy constructor and assignment \(&#39;Rule of Two&#39;\)
* 1 for Local variable hides global variable
* 1 for Use of potentially dangerous function
* 1 for Too few arguments to formatting function
* 1 for Short global name

**fixed alerts:**

* 20 for Wrong type of arguments to formatting function
* 3 for FIXME comment
* 1 for Missing header guard
* 1 for Function declared in block
* 1 for Too few arguments to formatting function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: core/cfg/cfg.cpp [hunk 1]
```diff
@@ -29,12 +31,23 @@ void savecfgf()
 }
 void  cfgSaveStr(const wchar * Section, const wchar * Key, const wchar * String)
 {
-	cfgdb.set(string(Section), string(Key), string(String));
+	const std::string section(Section);
+	const std::string key(Key);
+	const std::string value(String);
+	if (cfgHasGameSpecificConfig())
+	{
+		if (cfgdb.get(section, key, "") == value)
```
[the remark this answers is not in the document]
Comment (flyinghead) [reply core/cfg/cfg.cpp]: Right, I could concatenate the section and key names in per-game configs or something similar

File: core/linux-dist/main.cpp [hunk 2]
```diff
@@ -178,7 +143,6 @@ void os_DoEvents()
 
 void os_SetWindowText(const char * text)
 {
-	printf("%s\n",text);
```
[the remark this answers is not in the document]
Comment (flyinghead) [reply core/linux-dist/main.cpp]: The speed stats every 2 sec are still printed in the console, but not shown in the window title

**Pull request merged**